### DISCUSSION TITLE
socks proxy configuration assistance - macOS

### DISCUSSION THREAD

**Discussion by carljmosca** [containers/podman#19354] [General]
Running podman 4.5.1 on macOS (M2 processor)

I am trying to use a socks5 proxy that is running on localhost:5000

If I do a podman machine ssh and set https_proxy=socks5://host.containers.internal:5000, I am able leverage the proxy for a registry push.

If I try the same from the macOS terminal, I have not been able to make the same work.

I have tried the same setting I used above after locahost:5000, 127.0.0.1:5000, along with address that the above resolves to.

I am guessing I am missing something obvious - any help would be appreciated.

**Comment by Luap99** [comment]
You have to set `https_proxy=socks5://host.containers.internal:5000` when you start the machine (`podman machine start`), then the machine should use the correct proxy env

**Reply by carljmosca** [reply]
Thank you @Luap99 - I had added https_proxy and http_proxy to /etc/profile and restarted the podman machine.  When I ssh back in I can see that these are set and I am able to push the image from within the machine.  What I cannot do is a podman push from the macOS terminal.  I think I read your answer three times before it clicked that I had not tried starting podman with the values set.  Unfortunately I am still stuck.

**Reply by Luap99** [reply]
editing /etc/profile will not effect systemd services AFAIK

if you start the machine with `https_proxy=socks5://host.containers.internal:5000 podman machine start` it should copy the env var correctly so that the podman service in VM can use it. Then it should work, at least this is my understanding from other issues in the past.

**Comment by carljmosca** [comment]
> editing /etc/profile will not effect systemd services AFAIK

Was setting it there to test (and then use) a podman push from inside the machine as I needed to preserve the setting for reuse.

> if you start the machine with `https_proxy=socks5://host.containers.internal:5000 podman machine start` it should copy the env var correctly so that the podman service in VM can use it. Then it should work, at least this is my understanding from other issues in the past.

This method seems to have no effect on my machine but it may well be something strange I am (or not) doing.  Same issue for me with Docker so it may well be something particular to my configuration.
